### PR TITLE
Fix for Travis Error in ftp_download

### PR DESCRIPTION
**Pull request by viskey98**
Fixes issue: #234 
The unittest.mock have mocks only for normal functions. So I have used a custom function (AsyncMock) to mock the coroutine (handler.download()) behaviour. 
This fix will successfully mock the ftp_download behaviour with the desired result, which can fix the travis error.
Please suggest any changes if required :)

### CODE REVIEW

File: tanner/tests/test_rfi_emulation.py [hunk 1]
```diff
@@ -19,8 +28,9 @@ def test_http_download_fail(self):
         path = 'http://foobarfvfd'
         filename = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNone(filename)
-
+        
     def test_ftp_download(self):
+        self.handler.download_file = AsyncMock(return_value = '10ddd27e6bab5c8e2aa356906d1b71e5')
```
Comment (afeena) [inline tanner/tests/test_rfi_emulation.py]: I'd better mock download_file_ftp function

File: tanner/tests/test_rfi_emulation.py [hunk 2]
```diff
@@ -1,8 +1,17 @@
 import asyncio
 import unittest
-
+from unittest import mock
 from tanner.emulators import rfi
 
+# Custom function to mock ascyncio coroutines
+def AsyncMock(*args, **kwargs):
```
Comment (afeena) [inline tanner/tests/test_rfi_emulation.py]: lowercase with words separated by underscores 
https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

**Comment by viskey98** [comment]
@afeena The changes suggested by you have been made, please look into it :)

File: tanner/tests/test_rfi_emulation.py [hunk 3]
```diff
@@ -19,10 +28,11 @@ def test_http_download_fail(self):
         path = 'http://foobarfvfd'
         filename = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNone(filename)
-
+        
     def test_ftp_download(self):
+        self.handler.download_file_ftp = async_mock(return_value = '10ddd27e6bab5c8e2aa356906d1b71e5')
         path = 'ftp://mirror.yandex.ru/archlinux/lastupdate'
-        data = self.loop.run_until_complete(self.handler.download_file(path))
+        data = self.loop.run_until_complete(self.handler.download_file_ftp(path))
```
Comment (afeena) [inline tanner/tests/test_rfi_emulation.py]: mock download_file_ftp function - run download_file function :) assert filenames

**Comment by viskey98** [comment]
@afeena I have done the changes suggested by you,  please let me know if any more changes are required :)

File: tanner/tests/test_rfi_emulation.py [hunk 4]
```diff
@@ -19,8 +18,9 @@ def test_http_download_fail(self):
         path = 'http://foobarfvfd'
         filename = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNone(filename)
-
+        
     def test_ftp_download(self):
+        self.handler.download_file_ftp = mock.MagicMock(return_value = '10ddd27e6bab5c8e2aa356906d1b71e5')
         path = 'ftp://mirror.yandex.ru/archlinux/lastupdate'
         data = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNotNone(data)
```
Comment (afeena) [inline tanner/tests/test_rfi_emulation.py]: I suggest you assert with exactly returned value for mocked function. It checks that mocked function was called and proper value was used in download_file function

File: tanner/tests/test_rfi_emulation.py [hunk 5]
```diff
@@ -19,11 +19,12 @@ def test_http_download_fail(self):
...
         filename = self.loop.run_until_complete(self.handler.download_file(path))
         self.assertIsNone(filename)
-
+        
     def test_ftp_download(self):
+        self.handler.download_file_ftp = mock.MagicMock()
         path = 'ftp://mirror.yandex.ru/archlinux/lastupdate'
         data = self.loop.run_until_complete(self.handler.download_file(path))
-        self.assertIsNotNone(data)
+        self.handler.download_file_ftp.assert_called_with(yarl.URL(path))
```
Comment (afeena) [inline tanner/tests/test_rfi_emulation.py]: just for information - we can also write the test which allows to check if we get proper result from the `download_file` function when using ftp.

**Comment by afeena** [comment]
@viskey98 We can check returned value too, to be sure that download_file use the right value, not a random one :D But I think current state of the test is also ok :)

**Pull request merged**